1.10.0-alpha metrics PrometheusCollector error

`
Caused by: java.lang.IllegalStateException: No collection info for handle, this is a bug in the OpenTelemetry SDK.
	at io.opentelemetry.sdk.metrics.SdkMeterProvider$LeasedMetricProducer.collectAllMetrics(SdkMeterProvider.java:158) ~[opentelemetry-sdk-metrics-1.10.0-alpha.jar:1.10.0]
	at io.opentelemetry.exporter.prometheus.PrometheusCollector.collect(PrometheusCollector.java:35) ~[opentelemetry-exporter-prometheus-1.10.0-alpha.jar:1.10.0]
	at io.prometheus.client.CollectorRegistry.collectorNames(CollectorRegistry.java:113) ~[simpleclient-0.12.0.jar:na]
	at io.prometheus.client.CollectorRegistry.register(CollectorRegistry.java:50) ~[simpleclient-0.12.0.jar:na]
	at io.prometheus.client.Collector.register(Collector.java:308) ~[simpleclient-0.12.0.jar:na]
	at io.prometheus.client.Collector.register(Collector.java:301) ~[simpleclient-0.12.0.jar:na]
	at io.opentelemetry.exporter.prometheus.PrometheusCollector$Factory.apply(PrometheusCollector.java:76) ~[opentelemetry-exporter-prometheus-1.10.0-alpha.jar:1.10.0]
	at io.opentelemetry.sdk.metrics.SdkMeterProvider.<init>(SdkMeterProvider.java:84) ~[opentelemetry-sdk-metrics-1.10.0-alpha.jar:1.10.0]
	at io.opentelemetry.sdk.metrics.SdkMeterProviderBuilder.build(SdkMeterProviderBuilder.java:140) ~[opentelemetry-sdk-metrics-1.10.0-alpha.jar:1.10.0]
`

Like above, the latest 1.10.0-alpha version occur error, but the 1.9.1-alpha version not. There is a bug from the new version?

--- a/exporters/prometheus/src/main/java/io/opentelemetry/exporter/prometheus/PrometheusCollector.java
+++ b/exporters/prometheus/src/main/java/io/opentelemetry/exporter/prometheus/PrometheusCollector.java
@@ -25,13 +25,31 @@ import java.util.List;
  */
 public final class PrometheusCollector extends Collector implements MetricReader {
   private final MetricProducer metricProducer;
+  private volatile boolean registered = false;
 
   PrometheusCollector(MetricProducer metricProducer) {
     this.metricProducer = metricProducer;
   }
 
+  /**
+   * This method is called in {@link Factory#apply(MetricProducer)}. {@link Collector#register()}
+   * triggers a call to {@link #collect()}, which throws an error because {@link
+   * MetricProducer#collectAllMetrics()} is not yet read to accept calls. To get around this, we
+   * have {@link #collect()} exit early until registration is complete.
+   */
+  @SuppressWarnings("TypeParameterUnusedInFormals")
+  @Override
+  public <T extends Collector> T register() {
+    T result = super.register();
+    this.registered = true;
+    return result;
+  }
+
   @Override
   public List<MetricFamilySamples> collect() {
+    if (!registered) {
+      return Collections.emptyList();
+    }
     Collection<MetricData> allMetrics = metricProducer.collectAllMetrics();
     List<MetricFamilySamples> allSamples = new ArrayList<>(allMetrics.size());
     for (MetricData metricData : allMetrics) {

--- a/exporters/prometheus/src/test/java/io/opentelemetry/exporter/prometheus/PrometheusCollectorTest.java
+++ b/exporters/prometheus/src/test/java/io/opentelemetry/exporter/prometheus/PrometheusCollectorTest.java
@@ -7,11 +7,13 @@ package io.opentelemetry.exporter.prometheus;
 
 import static io.opentelemetry.api.common.AttributeKey.stringKey;
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatCode;
 import static org.mockito.Mockito.when;
 
 import com.google.common.collect.ImmutableList;
 import io.opentelemetry.api.common.Attributes;
 import io.opentelemetry.sdk.common.InstrumentationLibraryInfo;
+import io.opentelemetry.sdk.metrics.SdkMeterProvider;
 import io.opentelemetry.sdk.metrics.data.AggregationTemporality;
 import io.opentelemetry.sdk.metrics.data.DoublePointData;
 import io.opentelemetry.sdk.metrics.data.DoubleSumData;
@@ -25,6 +27,7 @@ import io.prometheus.client.exporter.common.TextFormat;
 import java.io.IOException;
 import java.io.StringWriter;
 import java.util.Collections;
+import java.util.concurrent.TimeUnit;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.ExtendWith;
@@ -41,6 +44,18 @@ class PrometheusCollectorTest {
     // Apply the SDK metric producer registers with prometheus.
     prometheusCollector = new PrometheusCollector(metricProducer);
     prometheusCollector.register();
+  }
+
+  @Test
+  void registerWithSdkMeterProvider() {
+    assertThatCode(
+            () ->
+                SdkMeterProvider.builder()
+                    .registerMetricReader(PrometheusCollector.create())
+                    .build()
+                    .forceFlush()
+                    .join(10, TimeUnit.SECONDS))
+        .doesNotThrowAnyException();
   }
 
   @Test

